script/Common.s.sol

```diff
@@ -340,6 +341,35 @@ abstract contract Common is Contracts, Params {
     rewardedActions = new RewardedActions();
   }
 
+  function _deployUniV3Pool() internal {
+    (address _token0, address _token1) = address(systemCoin) < address(collateral[WETH])
+      ? (address(systemCoin), address(collateral[WETH]))
+      : (address(collateral[WETH]), address(systemCoin));
+
+    uint256 _ethUSDPrice = delayedOracle[WETH].read();
+
+    address _uniV3Pool = INonfungiblePositionManager(NONFUNGIBLE_POSITION_MANAGER).createAndInitializePoolIfNecessary({
+      token0: _token0,
+      token1: _token1,
+      fee: HAI_POOL_FEE_TIER,
+      sqrtPriceX96: uint160(_ethUSDPrice) // REVIEW: How to calculate sqrtPriceX96 on-chain?
+    });
+
+    for (uint256 _i; _i < HAI_POOL_OBSERVATION_CARDINALITY;) {
+      IUniswapV3Pool(_uniV3Pool).increaseObservationCardinalityNext(500);
+      _i += 500;
+    }
+
+    // REVIEW: systemCoinOracle was set to be a HardcodedOracle during setupEnvironment(),
+    //         and will be set to be a DeviatedOracle during setupPostEnvironment()
```

Comment:
we no longer need Hardcoded or Deviated oracles :)
i'd leave them for Gorli tho, and add the deployment of the pool on mainnet

Reply:
I left the `HardcodedOracle` and the `DeviatedOracle` setup on Goerli deployment, where there's no UniV3 presence at the moment: https://github.com/Uniswap/v3-core/issues/609.

I left the `HardcodedOracle` setup on Mainnet deployment, since otherwise the `deployUniV3Relayer()` transaction reverts during `setupEnvironment()`. However, the same doesn't happen during `setupPostEnvironment()`.

Reply:
In order to stop the reversion caused by the PID initialization, it is no longer triggered during common deployment routine. This has permitted to avoid using the `HardcodedOracle` on Mainnet setup.